### DISCUSSION TITLE
Published site management

### DISCUSSION THREAD

**Discussion by Crims-on** [plentico/plenti#345] [Q&A] [answered]
Hello, I saw this project and it sounds like a perfect fit for my use case. I develop websites in Sveltekit for clients and this tool would make it easy for me and for clients to manage the website. Looking at the project and videos about it I found it unclear where the editing panel pops up. Can I edit content on an already deployed website?

**Comment by jimafisk** [comment, marked as answer]
Hi @Crims-on,

Thanks for your interest in Plenti! These are great questions, I'll try to clarify:

The CMS / editing panel is pretty tightly coupled with the actual framework for building sites. Plenti is meant to be a static site generator (SSG) or [Jamstack](https://jamstack.org/) website builder. If you're unfamiliar with those concepts, it just means that it produces HTML, CSS, and JS that can be deployed to a flat file hosting service like [GitLab pages](https://docs.gitlab.com/ee/user/project/pages/). That means you don't need to run a server or manage a backend programming language like NodeJS or Go anywhere.

It probably would be difficult to lift just the CMS out of this framework and integrate it into a Sveltekit project, although it's open source and you're welcome to do so if you'd like to try. The challenge is that the CMS relies on things that the SSG produces to tell the CMS how to behave (which field widgets to render, where the media lives, how to save content changes, etc). So by decoupling these things, you'd have to modify the CMS to get these instructions by another means, or modify your project to send these instructions.

On the other hand, Plenti is simply made up of svelte templates, so you could potentially migrate your site from Sveltekit to Plenti in order to take advantage of the CMS. Just make sure you understand what SSG / Jamstack is and that it's right for your particular use case before investing time doing that. Our company has built a couple of Plenti with a [Pocketbase](https://pocketbase.io/) backend to do "fullstack" things like commerce, user auth, private data, etc, so that's still definitely is possible, but it's also a lot of work.

> I found it unclear where the editing panel pops up

If your site is a Plenti site, it will automatically create a prop called `user` that you can simply call `$user.login()` to get the editing panel to appear. If you're doing local development, this will log you in without asking for any credentials. On a remote/deployed site, you'll need to set up an OAuth app that allows you to login into to your git backend (GitLab, Gitea, etc). There are some details in the docs (https://plenti.co/docs/cms-setup) but happy to discuss more if you end up going down that route and have questions.

Best of luck with your projects!

**Reply by Crims-on** [reply]
Thank you so much, and great work on the project, I look forward to using it!
